### PR TITLE
Add support for REQUEST_MESSAGE message

### PR DESCRIPTION
**Pull request by peterbarker**
Basically an alternative to requesting a message stream for a message.

This sort of message would become redundant:
```


      <entry value="520" name="MAV_CMD_REQUEST_AUTOPILOT_CAPABILITIES">
        <description>Request autopilot capabilities</description>
        <param index="1">1: Request autopilot version</param>
        <param index="2">Reserved (all remaining params)</param>
      </entry>
      <entry value="521" name="MAV_CMD_REQUEST_CAMERA_INFORMATION">
        <description>Request camera information (CAMERA_INFORMATION).</description>
        <param index="1">0: No action 1: Request camera capabilities</param>
        <param index="2">Reserved (all remaining params)</param>
      </entry>
      <entry value="522" name="MAV_CMD_REQUEST_CAMERA_SETTINGS">
        <description>Request camera settings (CAMERA_SETTINGS).</description>
        <param index="1">0: No Action 1: Request camera settings</param>
        <param index="2">Reserved (all remaining params)</param>
      </entry>
```

### CODE REVIEW

**Comment by peterbarker** [comment]
See this upstream mavlink pull request for the message: https://github.com/mavlink/mavlink/pull/1079

File: Tools/autotest/common.py [hunk 1]
```diff
@@ -1974,6 +1977,21 @@ def test_set_message_interval(self):
...
+        self.run_cmd(mavutil.mavlink.MAV_CMD_REQUEST_MESSAGE,
+                     180,
+                     0,
+                     0,
+                     0,
+                     0,
+                     0,
+                     0,
+                     timeout=timeout)
+        m = self.mav.recv_match(type='CAMERA_FEEDBACK', blocking=True, timeout=1)
```
Comment (OXINARF) [inline Tools/autotest/common.py]: Since this has a timeout, shouldn't it check that m is not None?

**Comment by peterbarker** [comment]
On Wed, 13 Feb 2019, Francisco Ferreira wrote:

> Since this has a timeout, shouldn't it check that m is not None?

Fixed.

**Comment by peterbarker** [comment]
Waiting for @amilcarlucas 's https://github.com/ArduPilot/mavlink/pull/90 to go in before this one.  I do have a branch with just the message in github.com/ardupilot/mavlink, but we should probably not merge that.

**Comment by peterbarker** [comment]
I've changed this PR to use a numeric constant in place of the `mavutil.mavlink` constant.

I believe the reason it fails CI is that we're installing MAVProxy after installing pymavlink; installing MAVProxy pulls in the pypi pymavlink, which doesn't have the new message in it.

This could be fixed, but if we do that then anybody wanting to run the autotest suite would need to install message definitions from our own mavlink sub-repository.  Taking @rmackay9 as a representative sample, that might be a tough ask :-)

**Comment by rmackay9** [comment]
ok, so I think we are happy to merge once this passes travis.

P.S. Personally I would raise an issue so we don't forget about the constant but it's up to you.

**Comment by peterbarker** [comment]
I've changed this PR to use a numeric constant in place of the `mavutil.mavlink` constant.

I believe the reason it fails CI is that we're installing MAVProxy after installing pymavlink; installing MAVProxy pulls in the pypi pymavlink, which doesn't have the new message in it.

This could be fixed, but if we do that then anybody wanting to run the autotest suite would need to install message definitions from our own mavlink sub-repository.  Taking @rmackay9 as a representative sample, that might be a tough ask :-)

**Pull request merged**